RefRed/reduction_table_handling/reduction_table_handler.py

```diff
@@ -39,73 +40,39 @@ def clear_rows_selected(self):
             self.__clear_plots()
         self.__clear_rows_big_table_data()
         self.__clear_rows_reduction_table()
-        self.__shifs_none_empty_rows_reduction_table()
+        self.__shift_none_empty_rows_reduction_table()
         self.__to_do_if_table_empty()
 
     def __to_do_if_table_empty(self):
         """If the table is now empty, various reset algos"""
-        _cell_value = str(self.parent.ui.reductionTable.item(0, 1).text())
+        _cell_value = str(self.table.item(0, 1).text())
         if _cell_value == "":
             self.__reset_default_config_file_name()
 
     def __clear_rows_reduction_table(self):
         _from_row = self.from_row
         _to_row = self.to_row
-        _nbr_col = self.parent.ui.reductionTable.columnCount()
+        _nbr_col = self.table.columnCount()
         for row_index in range(_from_row, _to_row + 1):
             for col_index in range(_nbr_col):
-                if col_index == 0:
```

Comment:
the new `__clear_reduction_table_cell` doesn't seem to be dealing with `col_index == 0`, which is a checkbox. What am I missing?

Reply:
This was another attempt at simplifying the logic and just clearing the state of the `QCheckBox` instead of recreating the checkbox, which also eliminates the need for reconnecting the `stateChanged` signal.

This is the part of `__clear_reduction_table_cell` that unchecks a checkbox in the table cell:
```
        checkbox_widget = self.__get_table_cell_checkbox_widget(row, col)
        if checkbox_widget:
            checkbox_widget.setChecked(False)
            checkbox_widget.setEnabled(True)
```